internal/condition/condition.go

```diff
@@ -217,7 +219,10 @@ func (e *EvaluableCondition) Evaluate(contextMaps ...map[string]*structpb.Value)
 		missingParameters = append(missingParameters, key)
 	}
 
-	out, details, err := e.celProgram.Eval(activation)
+	evalCtx, cancel := context.WithTimeout(ctx, time.Millisecond)
```

Comment:
side note - if we're going to set an aggressive timeout here, then we should have some metrics calculating the percentiles so that we can make a more informed decision what this value should be. I'll capture this work in a follow up PR since I'm focusing on some metrics right now.

However, in the spirit of making a more informed decision here, can we do a short and sweet analysis of how long a CEL evaluation takes in this code path `EvaluableCondition#Evaluate` if we evaluate a condition that is of the maximum evaluation cost (100 by default right now)?

That is, if we evaluate a condition with cost 100 thousands of times, what is the p99 value of that? Then we should use that value here. Otherwise this value may be more aggressive than it should be and could lead to server errors for conditions even if they are well below the 100 cost limit.

Reply:
You may consider using a TDigest in a benchmark to compute these quantiles/percentiles in a Go benchmark itself.

https://github.com/influxdata/tdigest


That would probs be the quickest/easiest way to get this data quickly.

Reply:
Updated PR description with benchmark results.